service/src/test/java/skills/intTests/UserPointsSpecs.groovy

```diff
@@ -1164,4 +1164,126 @@ class UserPointsSpecs extends DefaultIntSpec {
         data4.userTag == ['tag1', 'tag2', 'tag4', 'tag5']
     }
 
+    def 'user project points maximum filter is exclusive' () {
+        skillsService.deleteProjectIfExist(projId)
+        def proj = createProject()
+        def subject = createSubject()
+        List<Map> skills = createSkills(4, 1, 1, 100)
+
+        skillsService.createProject(proj)
+        skillsService.createSubject(subject)
+        skillsService.createSkills(skills)
+
+        def users = getRandomUsers(4)
+        skillsService.addSkill(skills[0], users[0], new Date())
+        skillsService.addSkill(skills[1], users[0], new Date())
+        skillsService.addSkill(skills[2], users[0], new Date())
+        skillsService.addSkill(skills[3], users[0], new Date())
+
+        skillsService.addSkill(skills[0], users[1], new Date())
+        skillsService.addSkill(skills[1], users[1], new Date())
+        skillsService.addSkill(skills[2], users[1], new Date())
+
+        skillsService.addSkill(skills[0], users[2], new Date())
+        skillsService.addSkill(skills[1], users[2], new Date())
+
+        skillsService.addSkill(skills[0], users[3], new Date())
+
+        when:
+        def projFull = skillsService.getProjectUsers(projId,  10, 1, "userId", true, "", 0, 100)
+        def proj76 = skillsService.getProjectUsers(projId,  10, 1, "userId", true, "", 0, 76)
+        def proj75 = skillsService.getProjectUsers(projId,  10, 1, "userId", true, "", 0, 75)
+        def proj50 = skillsService.getProjectUsers(projId,  10, 1, "userId", true, "", 0, 50)
+        def proj25 = skillsService.getProjectUsers(projId,  10, 1, "userId", true, "", 0, 25)
+
+        then:
+        projFull.count == 4
+        proj76.count == 3
+        proj75.count == 2
+        proj50.count == 1
+        proj25.count == 0
+
+    }
+
+    def 'user subject points maximum filter is exclusive' () {
+        skillsService.deleteProjectIfExist(projId)
```

Comment:
Please enhance the test to include at least two subjects with both distinct and overlapping users.